src/periodic.jl

```diff
@@ -0,0 +1,34 @@
+const PeriodicFunction = Union{typeof(cos), typeof(sin)}
+const PeriodicInput = Union{Real,TimeType}
+
+"""
+    Periodic(f, period, phase_shift) <: Transform
+
+Applies a periodic function `f` with provided `period` and `phase_shift` to the data.
+
+# Fields
+* `f::PeriodicFunction`: the periodic function
+* `period<:PeriodicInput`: the duration it takes for the periodic function to repeat
+* `phase_shift<:PeriodicInput`: adjusts the phase of the periodic function. A positive
+    value translates the function to the left, toward lower input values.
+"""
+struct Periodic{T<:PeriodicInput} <: Transform
+    f::PeriodicFunction
+    period::T
+    phase_shift::T
+end
+
+"""
+    Periodic(f::PeriodicFunction, period<:PeriodicInput) -> Periodic
+
+A constructor for [`Periodic`](@ref).
+Returns a `Periodic` transform with zero phase shift.
+"""
+function Periodic(f, period)
+    return Periodic{typeof(period)}(f, period, zero(typeof(period)))
+end
+
+function _apply!(x::AbstractArray{T}, P::Periodic; kwargs...) where T <: Real
+    x[:] = P.f.(2π .* x ./ P.period .+ P.phase_shift)
```

Comment:
Adding phase shift seems to be the more common convention (e.g. [Wikipedia article](https://en.wikipedia.org/wiki/Sine_wave)), but I think the old FeatureEngineering function effectively subtracts?

Reply:
Do they perform the same? If they do what does benchmarking say is more performant?

Reply:
I don't see why they would differ in performance, but they will return different values, because it changes the direction that the function is shifted. Whichever direction we choose can be reversed by passing a negative value for `phase_shift`. So my question relates more to what our users expect phase shift to do. 

I think I will change it to subtract, because (a) one can just change the sign of the phase shift argument to match their expectations (b) phase shift usually isn't needed for feature engineering purposes (c) it aligns with our current implementation so there's less confusion/updating usage.